carbonmark-data/components/charts/helpers/Configuration.tsx

```diff
@@ -1,7 +1,14 @@
-interface ChartConfigurationItem<T> {
-  id: Extract<keyof T, string>; // id of chart, also the key used to find data for this configuration item
+export interface ChartConfigurationItem<T> {
+  id: Extract<T, string>; // id of chart, also the key used by recharts to find data for this configuration item
   label: string; // label
   color: string; // color on the chart
   legendOrder?: number; // The order in which the element are displayd in the legend
 }
-export type ChartConfiguration<T> = Array<ChartConfigurationItem<T>>;
+/*
+A chart configuration is an array of configuration items that also bear data about how to query data (Q)
+*/
+export type ChartConfiguration<Q, M, C> = Array<{
+  query?: Q;
```

Comment:
Why is query nullable? I can see that we throw errors if it's undefined so why not enforce it to be non-null?

Reply:
It is not mandatory for the HistocalPriceCard: https://github.com/KlimaDAO/klimadao/pull/1487/files#diff-68aa516936992eed7a72a89d9182141ae44df7e7fcaadb3ba51bb078949870e1

Truth be told, it does not felt good when I made it optional. If you think of a better design. Do not hesitate.